### DISCUSSION TITLE
Issue with Namespaces in .d.ts File

### DISCUSSION THREAD

**Discussion by WINOFFRG** [egoist/tsup#870] [Q&A]
Hello! 

I am working on a UI Library which uses [Shaka Player](https://www.npmjs.com/package/shaka-player) as its dependency. This requirement is optionally installed by user on a specific version. And in my library I do some checks that if user has asked to use this library then use it.

So! For the types I have `types/shaka.d.ts` which 
```ts
declare module 'shaka-player' {
	export = shaka;
}
```

Creates the namespace for my project and exposes the typings! For `import shaka from 'shaka-player';` 
The types ae exposed from 
`\node_modules\.pnpm\shaka-player@4.3.3\node_modules\shaka-player\dist\shaka-player.compiled.d.ts`

While using `tsup src --dts` I am getting error

```
Error: namespace child (hoisting) not supported yet

  1 | declare module 'shaka-player' {
> 2 |   export = shaka;
    |   ^^^^^^^^^^^^^^^
  3 | }
  4 |
    at DeclarationScope.convertNamespace (\node_modules\.pnpm\tsup@6.7.0_typescript@5.0.2\node_modules\tsup\dist\rollup.js:7021:19)
```

New to `tsup` not sure where the issue is! Kindly help

**Comment by clayrisser** [comment]
I'm having this same issue.

**Comment by moltar** [comment]
It's an issue with `rollup-plugin-dts`, and unfortunately it was deemed "wontfix".

https://github.com/Swatinem/rollup-plugin-dts/issues/162

**Comment by ctlinker** [comment]
2025 and still dealing with this
